check_msid_status.py

```diff
@@ -13,102 +13,82 @@
 #-------------------------------------------------------------------------------
 #-- check_status: check status of msid                                        --
 #-------------------------------------------------------------------------------
-import traceback
 
 def check_status(msid, val, ldict, vdict):
     """
     check status of msid
     input:  msid    --- msid
             val     --- current value of msid
-            ldict   --- dictionary of limits for nemeric entry
+            ldict   --- dictionary of limits for numeric entry
             vdict   --- dictionary of msid <---> current value
     output: status  --- current status of msid
     """
 #
-#--- the value is nemeric
+#--- If value is numeric, then convert
 #
     try:
         val = float(val)
-        try:
-            status = check_status_neumeric(msid, val, ldict)
-            return status
-        except:
-            return 'GREEN'
+    except:
+        pass
 #
-#--- the value is letters
+#--- General Violation
 #
-    except:
-        if val in ['NaN', 'nan', '']:
-            return 'CAUTION'
-
-        elif msid in ['AOCONLAW', 'AOCPESTL', '4OBAVTMF', '4OBTOORF', 'COSCS128S',\
-                    'COSCS129S','COSCS130S', 'COSCS131S','COSCS132S','COSCS133S',\
-                    'COSCS107S','CORADMEN', 'CCSDSTMF', 'ACAFCT', 'AOFSTAR',\
-                    '2SHLDART', 'PLINE03T', 'PLINE04T', 'AACCCDPT', '3LDRTNO']:
-            try:
-                status = check_status_letter(msid, val, vdict)
-                return status
-            except KeyError:
-                return 'GREEN'
-            except:
-                traceback.print_exc()
-                return 'GREEN'
-        else:
-            if val in ['ERR', 'FALT', 'FAIL']:
-                return "CAUTION"
-            else:
-                return 'GREEN'
+    if val in ['ERR', 'FALT', 'FAIL', 'NaN', 'nan', '']:
+        return 'CAUTION'
+#
+#--- Edge Case Violation
+#
+    elif msid in ['AOCONLAW', 'AOCPESTL', '4OBAVTMF', '4OBTOORF', 'COSCS128S',\
+                  'COSCS129S','COSCS130S', 'COSCS131S','COSCS132S','COSCS133S',\
+                  'COSCS107S','CORADMEN', 'CCSDSTMF', 'ACAFCT', 'AOFSTAR', '2SHLDART']:
+        return check_status_edge_case(msid, val, vdict)
+#
+#--- Numerical Violation
+#
+    elif isinstance(val, float):
+        return check_status_numeric(msid, val, ldict)
+#
+#--- Unchecked Instance
+#
+    else:
+        return 'GREEN'
```

Comment:
Status Check is now organized to convert to a float if possible( numeric value), and then iterate over a checklist of edge cases. If the value is ERR, FAIL, etc. or the msid fits into one of the edge cases, then the status check is assigned according to that edge case. If none of this is met and the value is numeric, then we read from the CSH_limit_table.json to determine the status.